proof/courier.go

```diff
@@ -467,6 +478,8 @@ func (h *HashMailBox) RecvAck(ctx context.Context, sid streamID) error {
 
 // CleanUp atempts to tear down the mailbox as specified by the passed sid.
 func (h *HashMailBox) CleanUp(ctx context.Context, sid streamID) error {
+	defer h.rawConn.Close()
```

Comment:
We need to move this `h.rawConn.Close()` out of the `CleanUp()` method, since that is being called for two mailboxes (sender+receiver). So it needs to be done in `DeliverProof` as a `defer` before cleaning up the mailboxes.